crawl/crawl.go

```diff
@@ -79,21 +80,32 @@ func sorter(entries <-chan entryData) <-chan entryData {
 
 // verify requests `get-proof-by-hash` for a given entry and calculates the
 // root hash based on the returned audit_path plus the entry's hash. It then
-// compares that against a currently-calculated root hash. This provides a way
-// to do partial consistency checks along the way to verifying a full root hash
-// from scratch. It also ensures that if the log is serving get-proof-by-hash
-// requests using precalculated hashes, those precalculated hashes have not
-// been corrupted.
+// compares that against a root hash calculated by another method. This
+// provides a way to do partial consistency checks along the way to verifying
+// a full root hash from scratch. It also ensures that if the log is serving
+// get-proof-by-hash requests using precalculated hashes, those precalculated
+// hashes have not been corrupted.
 func verify(logClient *client.LogClient, e entryData, rootHash []byte) error {
 	currentTreeSize := e.index + 1
-	pbh, err := logClient.GetProofByHash(context.Background(), e.merkleLeafHash, uint64(currentTreeSize))
-	if err != nil {
-		return fmt.Errorf("getting proof by hash: %w", err)
+
+	var pbh *ct.GetProofByHashResponse
+	var err error
+	for {
+		pbh, err = logClient.GetProofByHash(context.Background(), e.merkleLeafHash, uint64(currentTreeSize))
+		if err != nil {
+			var netError net.Error
+			if errors.As(err, &netError) && netError.Timeout() {
+				log.Printf("get-proof-by-hash: %s", err)
+				time.Sleep(3 * time.Second)
+				continue
+			}
+			return fmt.Errorf("getting proof by hash: %w", err)
+		}
+		break
```

Comment:
Since we're using a bunch of trillian stuff anyway, probably easier to use https://github.com/google/trillian/blob/master/client/backoff/backoff.go, the same lib that scanner/fetcher uses.